packages/flutter/test/painting/fake_codec.dart

```diff
@@ -44,8 +44,8 @@ class FakeCodec implements ui.Codec {
   @override
   Future<ui.FrameInfo> getNextFrame() {
     _numFramesAsked += 1;
-    final SynchronousFuture<ui.FrameInfo> result =
-      SynchronousFuture<ui.FrameInfo>(_frameInfos[_nextFrame]);
+    final Future<ui.FrameInfo> result =
+      Future<ui.FrameInfo>.value(_frameInfos[_nextFrame]);
```

Comment:
Is `SynchronousFuture` broken, or does it simply not suitable for this use-case because it's too eager to call the `then` callback? (or something else?)

Reply:
There's actually some active discussion going on about SynchronousFuture here:
https://github.com/flutter/flutter/issues/151579#issuecomment-2247128158

I personally don't know the history of why SynchronousFuture was added in the first place. It provides a very different behavior than a normal Future and doesn't really conform to the Dart spec. 

I don't know that its "broken" though... it does what it seems like it was designed to do. But in this specific case it definitely causes an incorrect render order given a fully valid implementation of Dart's async semantics. This same bug would also occur if the test were run with dart2js.

Reply:
Gotcha.